tracer/test/Datadog.Trace.Tests/Debugger/SpanCodeOriginTests.cs

```diff
@@ -0,0 +1,118 @@
+// <copyright file="SpanCodeOriginTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Configuration.Telemetry;
+using Datadog.Trace.Debugger;
+using Datadog.Trace.Debugger.SpanCodeOrigin;
+using Datadog.Trace.VendoredMicrosoftCode.System.Collections.Immutable;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Debugger
+{
+    public class SpanCodeOriginTests
+    {
+        private const string CodeOriginTag = "_dd.code_origin";
+
+        [Fact]
+        public void SetCodeOrigin_WhenSpanIsNull_DoesNotThrow()
+        {
+            // Should not throw
+            SpanCodeOriginManager.Instance.SetCodeOrigin(null);
+        }
+
+        [Fact]
+        public void SetCodeOrigin_WhenDisabled_DoesNotSetTags()
+        {
+            // Arrange
+            CreateCodeOriginManager();
+
+            var span = new Span(new SpanContext(1, 2, SamplingPriority.UserKeep), DateTimeOffset.UtcNow);
+
+            // Act
+            SpanCodeOriginManager.Instance.SetCodeOrigin(span);
+
+            // Assert
+            Assert.Null(span.Tags.GetTag(CodeOriginTag + ".type"));
+        }
+
+        [Fact]
+        public void SetCodeOrigin_WhenEnabled_SetsCorrectTags()
+        {
+            // Arrange
+            CreateCodeOriginManager(true);
+
+            var span = new Span(new SpanContext(1, 2, SamplingPriority.UserKeep), DateTimeOffset.UtcNow);
+
+            // Act
+            TestMethod(span);
+
+            // Assert
+            Assert.NotNull(span.Tags.GetTag($"{CodeOriginTag}.type"));
+            Assert.Equal("exit", span.Tags.GetTag($"{CodeOriginTag}.type"));
```

Comment:
nit: using `Should()` gives better error methods than using `Assert.Equal` etc. Also this will fail on the first failure, whereas you can use `Should()` to see _everything_ that's wrong in one error message 🙂

Reply:
Fix in ccc2d563e115d67c5eb0463aae6030070b923cef

Reply:
Fix in ccc2d563e115d67c5eb0463aae6030070b923cef